### ISSUE TITLE
[BUG] Android - Module was compiled with an incompatible version of Kotlin. 

### ISSUE THREAD

**Issue by MasonLe2497** [mrousavy/react-native-blurhash#144]
# Bug

**Describe the bug**

**To Reproduce**
Install blurhash
build android

**Expected behavior**
Build android succeedfully

**Screenshots**
![image](https://user-images.githubusercontent.com/43195241/149481000-4c74a315-c2cb-41d4-9df9-8806b2e4d9f0.png)

# Environment

**Additional context**

```
System:
    OS: macOS 11.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 146.59 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.0.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.11.2 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0
      System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7583922
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.4 => 0.66.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

```json
"react": 17.0.2,
"react-native": 0.66.4,
"react-native-blurhash": 1.1.8
```

**Comment by andresribeiro** [comment]
i'm having the same problem

**Comment by Yonom** [comment]
This also affects us, we downgraded to 1.1.5, which is the highest working version.

Similar issue: https://github.com/expo/expo/issues/15989#issuecomment-1019095115

**Comment by MasonLe2497** [comment]
I patch-package with RNBlurhash_kotlinVersion=1.6.10 then it working

**Comment by Dag0n** [comment]
looks like the npm version is a little beind this.

If you change your package.json from 
`"react-native-blurhash": "^1.1.8",`
to 
`"react-native-blurhash": "https://github.com/mrousavy/react-native-blurhash/",`

it will run as expected.

**Comment by JacobJaffe** [comment]
Yeah, it looks like https://github.com/mrousavy/react-native-blurhash/pull/150 fixes this?

**Comment by jonxssc** [comment]
Bug is still there, i have a temporary fix for everyone who faces this error:
go to `react-native-blurhash` in the `node-modules` folder. There go to `android/build.gradle` and search for this line `org.jetbrains.kotlin:kotlin-gradle-plugin`, change it to: `org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.20` save, then run `cd android && ./gradlew clean` and then restart your App, works for me, i hope this bug gets fixed soon

**Comment by andybui090** [comment]
For me, adding config kotlinVersion in file build.gradle as 
buildscript {
    ext {
        buildToolsVersion = "33.0.0"
        minSdkVersion = 23 //set min sdk to support good
        compileSdkVersion = 33
        targetSdkVersion = 33

        kotlinVersion = "1.7.10" //fix kotlin version RN blurhash
        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
        ndkVersion = "23.1.7779620"
    }
}
